R/dropDuplicateIDs.R

```diff
@@ -0,0 +1,78 @@
+
+
+#' Drop duplicate IDs in a \code{GADSdat}.
+#'
+#' Drop rows with duplicate IDs in a \code{GADSdat} object based on numbers of missing values.
+#'
+#' If duplicate IDs occur, it is often desirable to keep the row with the least missing information.
+#' Therefore, \code{dropDuplicateIDs} dops rows based on number of missing values
+#' on the specified variables (\code{varNames}).
+#'
+#' If multiple rows have the same number of missing values, a warning is issued
+#' and the first of the respective rows is kept.
+#'
+#'@param GADSdat A \code{GADSdat} object.
+#'@param ID Name of the ID variable.
+#'@param varNames Character vector of variable names: Sum of missing values
+#' on these variables decide which rows are kept. Per default, all variables
+#' except the ID variable are used.
+#'
+#'@return Returns the \code{GADSdat} with duplicate ID rows removed.
+#'
+#'@examples
+#' # create example data set
+#' gads_ori <- import_DF(data.frame(id_var = c(1, 2, 5, 4, 4),
+#'   var1 = c(1, 2, -99, 1, -99)))
+#' gads_ori <- changeMissings(gads_ori, varName = "var1",
+#'   value = -99, missings = "miss")
+#'
+#' # drop duplicate IDs
+#' dropDuplicateIDs(gads_ori, ID = "id_var")
+#'
+#'@export
+dropDuplicateIDs <- function(GADSdat, ID, varNames = setdiff(namesGADS(GADSdat), ID)) {
+  UseMethod("dropDuplicateIDs")
+}
+
+#'@export
+dropDuplicateIDs.GADSdat <- function(GADSdat, ID, varNames = setdiff(namesGADS(GADSdat), ID)) {
+  check_GADSdat(GADSdat)
```

Comment:
Tiny nit: I was trying to check these line-by-line and was wondering about the NULL returns for `check_GADSdat()` and `check_vars_inGADSdat()`, but not in `check_single_varName()`. Maybe, there could also be `invisible(return())`'s for these calls in case of no errors
?

Reply:
Modified as suggested!

Reply:
> Maybe @liebelta, @OtteKatrin and @burbliesj can state their most common use case?

The request for this function comes from the parent questionnaire. Here, sometimes there are questionnaires from both parents. This means that two parent questionnaires are assigned to one student. However, we only need one questionnaire per student and therefore have to remove the questionnaire of the parent with more missings.

Reply:
In the current data, there don´t seem to be duplicate IDs. Will come back here as soon as final datasets are available.